metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -181,10 +182,32 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
         """A list of functions that transforms the workunits produced by this source.
         Run in order, first in list is applied first. Be careful with order when overriding.
         """
+        browse_path_processor: Optional[MetadataWorkUnitProcessor] = None
+        if (
+            self.ctx.pipeline_config
+            and self.ctx.pipeline_config.flags
+            and self.ctx.pipeline_config.flags.get("generate_browse_path_v2")
+        ):
+            platform = getattr(self, "platform", None) or getattr(
+                self.get_config(), "platform", None
+            )
+            env = getattr(self.get_config(), "env", None)
+            browse_path_drop_dirs = [
+                platform,
+                platform and platform.lower(),
+                env,
+                env and env.lower(),
+            ]
+            browse_path_processor = partial(
+                auto_browse_path_v2,
+                [s for s in browse_path_drop_dirs if s is not None],
```

Comment:
does it work to simply drop `n` elements from the original browse path - why do we have this whole dynamic system here?

Reply:
Sometimes we include env + platform, sometimes just platform